src/main/java/org/apache/maven/plugins/assembly/artifact/DefaultDependencyResolver.java

```diff
@@ -130,10 +130,8 @@ void updateModuleSetResolutionRequirements( ModuleSet set, DependencySet depende
             {
                 if ( p.getArtifact() == null )
                 {
-                    // TODO: such a call in MavenMetadataSource too - packaging not really the intention of type
-                    final Artifact artifact =
-                        resolver.createArtifact( p.getGroupId(), p.getArtifactId(), p.getVersion(), p.getPackaging() );
-                    p.setArtifact( artifact );
+                    p.setArtifact( artifactFactory.createBuildArtifact( p.getGroupId(), p.getArtifactId(),
```

Comment:
`ArtifactFactory` is also deprecated, I would like to try `new DefaultArtifact(...)`